src/bson_mapper/bson_streambuf.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright 2016 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cassert>
+#include <cstdio>
+#include <iostream>
+#include <streambuf>
+
+#include <mongocxx/client.hpp>
+#include <mongocxx/instance.hpp>
+
+#include "bson_streambuf.hpp"
+
+#include <bson_mapper/config/prelude.hpp>
+
+namespace bson_mapper {
+BSON_MAPPER_INLINE_NAMESPACE_BEGIN
+
+bson_output_streambuf::bson_output_streambuf(mongocxx::collection *coll) {
```

Comment:
You can use a [member initializer list](http://en.cppreference.com/w/cpp/language/initializer_list) here.